kubernetes/windows/main.ps1

```diff
@@ -311,6 +311,39 @@ function Set-EnvironmentVariables {
         Write-Host "Failed to set environment variable KUBERNETES_PORT_443_TCP_PORT for target 'machine' since it is either null or empty"
     }
 
+if (![string]::IsNullOrEmpty($isAADMSIAuth) -and $isAADMSIAuth.ToLower() -eq 'true') {
+    [System.Environment]::SetEnvironmentVariable("MONITORING_DATA_DIRECTORY", "C:\\opt\\genevamonitoringagent\\datadirectory", "Process")
+    [System.Environment]::SetEnvironmentVariable("MONITORING_MCS_MODE", "1", "Process")
+    [System.Environment]::SetEnvironmentVariable("MONITORING_ROLE_INSTANCE", "cloudAgentRoleInstanceIdentity", "Process")
+    [System.Environment]::SetEnvironmentVariable("MCS_AZURE_RESOURCE_ENDPOINT", "https://monitor.azure.com/", "Process")
```

Comment:
This has to be cloud specific. For  example for government cloud, this has to be https://monitor.azure.us/ and for China cloud, https://monitor.azure.cn/ etc.